src/components/DateRangePicker/SelectionPane.js

```diff
@@ -0,0 +1,128 @@
+const React = require('react');
+const moment = require('moment');
+const PropTypes = require('prop-types');
+
+const DefaultRanges = require('../DateRangePicker/DefaultRanges');
+
+const StyleConstants = require('../../constants/Style');
+
+class SelectionPane extends React.Component {
+  static propTypes = {
+    currentDate: PropTypes.string,
+    defaultRanges: PropTypes.array,
+    handleDateBoxClick: PropTypes.func,
+    isLargeOrMediumWindowSize: PropTypes.bool,
+    primaryColor: PropTypes.string,
+    selectedBox: PropTypes.string,
+    selectedEndDate: PropTypes.number,
+    selectedStartDate: PropTypes.number,
+    setCurrentDate: PropTypes.func
+  };
+
+  state = {
+    selectedBox: 'from'
+  }
+
+  _handleDateBoxClick = (date, selectedBox) => {
+    this.props.handleDateBoxClick(date, selectedBox);
```

Comment:
I believe we settled on using the `onDoSomething` style for the prop name and the imperative `doSomething` name for the function passed in for prop handlers. Correct me if I'm wrong though, I'm always mixing those up.